src/mbgl/sprite/sprite_atlas.hpp

```diff
@@ -114,35 +105,18 @@ class SpriteAtlas : public util::noncopyable {
 
     SpriteAtlasObserver* observer = nullptr;
 
-    // Lock for sprites and dirty maps.
-    std::mutex mutex;
-
-    // Stores all current sprites.
-    Sprites sprites;
-
-    // Stores all Sprite IDs that changed since the last invocation.
-    Sprites dirtySprites;
-
-    struct Holder : private util::noncopyable {
-        Holder(std::shared_ptr<const SpriteImage>, Rect<uint16_t>);
-        Holder(Holder&&);
+    struct Entry {
         std::shared_ptr<const SpriteImage> spriteImage;
-        const Rect<uint16_t> pos;
+        optional<Rect<uint16_t>> iconRect;
+        optional<Rect<uint16_t>> patternRect;
```

Comment:
What is the difference between iconRect and patternRect? This could benefit from a comment